tests/test_data_source.py

```diff
@@ -3,13 +3,16 @@
 import pytest
 from snakeoil import compression, data_source
 from snakeoil.osutils import pjoin
-from snakeoil.test.fixtures import TempDir
 
 
-class TestDataSource(TempDir):
+class TestDataSource:
 
     supports_mutable = True
 
+    @pytest.fixture(autouse=True)
+    def _setup(self, tmp_path):
+        self.dir = str(tmp_path)
```

Comment:
It looks to hard to use correctly here `tmp_path` instead of `self.dir` (because of the various overriding of `get_obj`), so only here I took  a shortcut.